src/Mvc/Mvc.Razor.RuntimeCompilation/src/DependencyInjection/RazorRuntimeCompilationMvcCoreBuilderExtensions.cs

```diff
@@ -78,6 +81,24 @@ internal static void AddServices(IServiceCollection services)
 
             services.TryAddSingleton<IViewCompilerProvider, RuntimeViewCompilerProvider>();
 
+            var actionDescriptorProvider = services.FirstOrDefault(f =>
```

Comment:
We already do this for the compiler, but it doesn't make it any better. I think we might need to make this type public.